kubetest/main.go

```diff
@@ -368,9 +370,23 @@ func findVersion() string {
 	return "unknown" // Sad trombone
 }
 
+// maybeMergeMetadata will add new keyvals into the map; quietly eats errors.
+func maybeMergeJSON(meta map[string]string, path string) {
+	if data, err := ioutil.ReadFile(path); err == nil {
+		json.Unmarshal(data, &meta)
+	}
+}
+
 // Write metadata.json, including version and env arg data.
-func writeMetadata(path string) error {
+func writeMetadata(o *options) error {
```

Comment:
Would slightly prefer you pass in the two needed values rather than all options. 

Also do you think this behavior is worth a unit test? We've been slowly increasing the coverage for this program.